pkg/db/db.go

```diff
@@ -0,0 +1,155 @@
+//Package db provides operations on SQLite DB
+package db
+
+import (
+	"database/sql"
+	"os"
+	// blank import to register sqlite3 driver
+	_ "github.com/mattn/go-sqlite3"
+)
+
+const dbDir string = "/var/lib/leapp/"
+const dbName string = "leapp.db"
+
+//Database type is sql.DB
+type Database sql.DB
+
+//Record type represents DB record description
+type Record struct {
+	UID, Name, Host, DataType, Data string
+}
+
+//EnsureDbPath checks wether the leapp directory exists, if not it tries to create it
+//temporary function, will be solved in packaging later
+func EnsureDbPath() {
+	if err := os.MkdirAll(dbDir, os.ModePerm); err != nil {
+		panic(err)
+	}
+}
+
+func (db *Database) sql() *sql.DB {
+	return (*sql.DB)(db)
+}
+
+//connect represents internal function for DB connection to specific DB (i.e. for test, we use in-memory DB)
+func connect(dbPath string) *Database {
+	conn, err := sql.Open("sqlite3", dbPath)
+	if err != nil {
+		panic(err)
+	}
+	db := (*Database)(conn)
+	return db
+}
+
+//Connect connects to DB
+func Connect() *Database {
+	conn := connect(dbDir + dbName)
```

Comment:
Please do use filepath.Join(dbDir, dbName) from the "path/filepath" package - This code can introduce bugs otherwise if someone changes for example the dbDir contant and forgets to add the pending slash